xdem/dem.py

```diff
@@ -0,0 +1,163 @@
+"""
+dem.py provides a class for working with digital elevation models (DEMs)
+"""
+import pyproj
+import warnings
+from geoutils.satimg import SatelliteImage
+from pyproj import Transformer
+
+def parse_vref_from_product(product):
+
+    # sources for defining vertical references:
+    # AW3D30: https://www.eorc.jaxa.jp/ALOS/en/aw3d30/aw3d30v11_format_e.pdf
+    # SRTMGL1: https://lpdaac.usgs.gov/documents/179/SRTM_User_Guide_V3.pdf
+    # SRTMv4.1: http://www.cgiar-csi.org/data/srtm-90m-digital-elevation-database-v4-1
+    # ASTGTM2/ASTGTM3: https://lpdaac.usgs.gov/documents/434/ASTGTM_User_Guide_V3.pdf
+    # NASADEM: https://lpdaac.usgs.gov/documents/592/NASADEM_User_Guide_V1.pdf !! HGTS is ellipsoid, HGT is EGM96 geoid !!
+    # ArcticDEM (mosaic and strips): https://www.pgc.umn.edu/data/arcticdem/
+    # REMA (mosaic and strips): https://www.pgc.umn.edu/data/rema/
+    # TanDEM-X 90m global: https://geoservice.dlr.de/web/dataguide/tdm90/
+    # COPERNICUS DEM: https://spacedata.copernicus.eu/web/cscda/dataset-details?articleId=394198
+
+    if product in ['ArcticDEM/REMA','TDM1','NASADEM-HGTS']:
+        vref = 'WGS84'
+    elif product in ['AW3D30','SRTMv4.1','SRTMGL1','ASTGTM2','NASADEM-HGT']:
+        vref = 'EGM96'
+    elif product in ['COPDEM']:
+        vref = 'EGM08'
+    else:
+        vref = None
+
+    return vref
+
+class DEM(SatelliteImage):
+
+    def __init__(self, filename, vref=None, vref_grid=None, ccrs = None, read_vref_from_prod=True, silent=False, **kwargs):
+
+        super().__init__(filename, **kwargs)
+
+        if self.nbands > 1:
+            raise ValueError('DEM rasters should be composed of only one band only')
+
+        # priority to user input
+        self.vref = vref
```

Comment:
Like the Raster._data attribute, this should be set to a hidden attribute self._vref with a getter/setter method to prevent users from setting it to anything. See geoutils.Raster for an example.

Reply:
In the end I didn't add it because I don't think the usage of vref is really comparable to data. It would not be used much more than an attribute such as nbands or crs in Raster for instance, for which there is no need for a getter/setter method